src/JSONSchema/path-handler.ts

```diff
@@ -0,0 +1,107 @@
+import { JSONSchemaType } from './types'
+import { useFormContext } from '../components'
+
+const concatFormPath = (path: string, newNode: string): string => {
+  return path + '/' + newNode
+}
+
+const getSplitPath = (path: string): Array<string> => {
+  const split = path.split('/')
+  if (split[0] === '#') {
+    split.shift()
+  }
+  if (split[split.length - 1] === '') {
+    split.pop()
+  }
+  return split
+}
+
+const useObjectFromPath = (path: string): [JSONSchemaType, boolean, string] => {
+  const splitPath = getSplitPath(path)
+  let currentOriginal = useFormContext().schema
+  let isRequired = false
+  let objectName = ''
+
+  for (let node = 0; node < splitPath.length; node++) {
+    if (
+      currentOriginal &&
+      currentOriginal.type &&
+      currentOriginal.type === 'object'
+    ) {
+      if (
+        currentOriginal.required &&
+        (currentOriginal.required as Array<string>).indexOf(splitPath[node]) >
+          -1
+      ) {
+        isRequired = true
+      } else {
+        isRequired = false
+      }
+      objectName = splitPath[node]
+      currentOriginal = currentOriginal.properties[splitPath[node]]
+    } else {
+      currentOriginal = {}
+      break
+    }
+  }
```

Comment:
This `for` loop seems a bit complex, maybe it could be a function

Reply:
I don't know, maybe? But the function is already majorly this loop, I don't know of a better way to navigate the JSON Schema